dev/com.ibm.ws.security.oauth/src/com/ibm/ws/security/oauth20/plugins/OidcBaseClientValidator.java

```diff
@@ -92,10 +95,45 @@ public OidcBaseClient validateCreateUpdate() throws OidcServerException {
         return this.client;
     }
 
+    /**
+     * check for disallowed characters that could allow javascript to be fed back to ui.
+     */
+    private void detectIllegalChars() throws OidcServerException {
+        detectCh(client.getClientId());
+        detectCh(client.getClientSecret());
+        detectCh(client.getRedirectUris());
+        detectCh(client.getClientName());
+        detectCh(client.getPostLogoutRedirectUris());
+        detectCh(client.getPreAuthorizedScope());
+        detectCh(client.getFunctionalUserId());
+        detectCh(client.getFunctionalUserGroupIds());
+    }
+
+    // detect illegal chars
+    private void detectCh(@Sensitive String s) throws OidcServerException {
+        if (s == null || s.length() == 0) {
+            return;
+        }
+        for (int i = 0; i < illegalChars.length; i++) {
+            if (s.contains(illegalChars[i])) {
+                String description = TraceNLS.getFormattedMessage(OidcBaseClientValidator.class,
```

Comment:
The `Tr.formatMessage()` method offers a less verbose interface for obtaining an NLS message and is typically what's used. I have a suspicion that the `TraceNLS.getFormattedMessage()` method was more of a legacy method brought over from the old WAS traditional code base. The new message lets you just do

```Java
private static TraceComponent tc = Tr.register(OidcBaseClientValidator.class, "OAUTH", MESSAGE_BUNDLE);
...
    Tr.formatMessage(tc, "OAUTH_CLIENT_REGISTRATION_ILLEGAL_CHAR", new Object[] { illegalChars[i] });
```

I suppose if you wanted to get fancy, the method could also find _all_ of the illegal characters in the string and include those in the error message as opposed to just the first instance.

Reply:
agreed, but I'd like to leave as is since it's consistent with every other usage of this within the class.